### PR TITLE
Specify builder for docker build

### PR DESCRIPTION
**Pull request by pingsutw**
## Why are the changes needed?
Failed to build the sandbox-bundled

```bash
(flytekit-3.10) ➜  make flyte
```

```python
docker buildx build --build-arg FLYTECONSOLE_VERSION=latest --platform linux/amd64 --tag flyte-binary:sandbox --output type=docker,dest=images/tar/amd64/flyte-binary.tar ../..
[+] Building 0.0s (0/0)                                                                                                                                                                                       docker:desktop-linux
ERROR: Docker exporter is not supported for the docker driver.
Switch to a different driver, or turn on the containerd image store, and try again.
Learn more at https:/docs.docker.com/go/build-exporters/
make: *** [flyte] Error 1
```

## What changes were proposed in this pull request?
Specify builder for docker build

## Check all the applicable boxes 

- [ ] I updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [ ] All commits are signed-off.

### CODE REVIEW

File: docker/sandbox-bundled/Makefile [hunk 1]
```diff
@@ -38,12 +47,7 @@ manifests: dep_update
 		kustomize/complete-agent > manifests/complete-agent.yaml
 
 .PHONY: build
-build: flyte dep_update manifests
-	[ -n "$(shell docker buildx ls | awk '/^flyte-sandbox / {print $$1}')" ] || \
-		docker buildx create --name flyte-sandbox \
-		--driver docker-container --driver-opt image=moby/buildkit:master \
-		--buildkitd-flags '--allow-insecure-entitlement security.insecure' \
-		--platform linux/arm64,linux/amd64
+build: create_builder flyte dep_update manifests
```
Comment (Future-Outlier) [inline docker/sandbox-bundled/Makefile:50]: Need some tiny change, cause `create_builder` will be called in `flyte`

Comment (Future-Outlier) [reply docker/sandbox-bundled/Makefile:50]: Should we remove repetitive called?

Comment (pingsutw) [reply docker/sandbox-bundled/Makefile]: done

**Review by Future-Outlier — APPROVED** [review]
LGTM, thank you

**Pull request merged**